app/dao/fact_notification_status_dao.py

```diff
@@ -239,7 +241,11 @@ def fetch_notification_status_for_service_for_day(bst_day, service_id):
 
 
 def fetch_notification_status_for_service_for_today_and_7_previous_days(service_id, by_template=False, limit_days=7):
-    start_date = midnight_n_days_ago(limit_days)
+    if limit_days == 1:
+        start_date = get_midnight(datetime.now(tz=pytz.utc))
+    else:
+        start_date = midnight_n_days_ago(limit_days)
```

Comment:
I think this will return the correct datetime for the case we are trying to fix, but I think there are a couple issues with this change:
- `start_date` is a timezone aware datetime if `limit_days = 1`, and it is timezone naive if `limit_days > 1`. That makes the code harder to understand
- the way `start_date` is calculated is very different if `limit_days = 1` vs if `limit_days > 1`. This might cause issues in the future when we forget this detail and we are calling the endpoint via notification-admin

Could we just always use utc midnight for this function? I'm thinking something like:
```
def utc_midnight_n_days_ago(number_of_days):
    """
    Returns utc midnight a number of days ago.
    """
    return get_midnight(datetime.utcnow() - timedelta(days=number_of_days))
```
in `app/utils.py`, and then:
```
start_date = utc_midnight_n_days_ago(limit_days)
```
in this file.

Reply:
Yeah for sure, I specifically targeted the 1 day case, but you make good points here.  Let's chat!